### ISSUE TITLE
Unification of the utility language?

### ISSUE THREAD

**Issue by treylav** [YoYoGames/GM-TestFramework#29]
Hello.

Do you have plans to unify scripting language of non-GML part of the project? Right now, NodeJS and Python are solving problems that are not so specific that you can't use just one, making distribution easier.

Thank you in advance.

**Comment by DiasFranciscoA** [comment]
If you believe you can contribute to enhancing the framework launcher, please feel free to submit a pull request (PR). However, it is crucial to ensure that any changes made do not modify or disrupt the existing API. As the internal framework we use, altering it at this stage would necessitate adjustments to other components in the pipeline. Nevertheless, if the proposed change is transparent and primarily affects the underlying language, there is a possibility that it could be incorporated and eventually become a standard feature.

**Comment by treylav** [comment]
Yes, I share your concerns, and this is holding me back from working on improving the framework launcher. Since the only way to check the correctness of refactoring at the moment is to launch the framework itself using the launcher in various conditions.

Perhaps the right way would be to try to cover the launcher itself with tests to begin with?

**Comment by DiasFranciscoA** [comment]
Being an open source project the community is free and encouraged to contribute with their own improvements by creating PRs.

The reason for the project to be written in two different languages relies mainly on the fact that initially the server was running constantly on a web hosted machine that required the usage of NodeJS. To make this project public we needed to create a standalone version of it. And it ended up being just a pure copy of the server code.

Ideally the code base should be written in Python. So, a pure NodeJS port won't be accepted as a PR as it diverges from our internal pipeline structure.

**Comment by YYDan** [comment]
Not sure why this wasn't closed when the answer above was given in 2023, but have done so now.
